SnaffCore/ActiveDirectory/AdData.cs

```diff
@@ -60,40 +78,38 @@ public AdData()
             {
                 Mq.Trace("Target DC and Domain specified: " + MyOptions.TargetDomain + " + " + MyOptions.TargetDc);
                 _targetDc = MyOptions.TargetDc;
-                _targetDomain = MyOptions.TargetDomain;
+                _targetDomain = Domain.GetDomain(new DirectoryContext(DirectoryContextType.Domain, MyOptions.TargetDomain));
             }
             // no target DC or domain set
             else
             {
                 Mq.Trace("Getting current domain from user context.");
-                _currentDomain = Domain.GetCurrentDomain();
-                _targetDomain = _currentDomain.Name;
-                _targetDc = _targetDomain;
+//                _currentDomain = Domain.GetCurrentDomain();
+//                _targetDomain = _currentDomain.Name;
+                _targetDomain = Domain.GetCurrentDomain();
```

Comment:
doing Domain.GetCurrentDomain() or Domain.GetDomain() is fine in the 'no target DC or domain set' path, but when the user is manually setting those as strings we need to avoid those calls as they do SRV lookups.